### DISCUSSION TITLE
subprocess error but it's work on Pydroid app?

### DISCUSSION THREAD

**Discussion by shaheen-coder** [termux/termux-app#3564] [Q&A] [answered]
When I try to run command ****pip install django-allauth**** it doesn't Install 
But it's working on Pydroid 3 app ? 

![Screenshot_2023-08-30-13-00-58-188_com.termux.jpg](https://github.com/termux/termux-app/assets/116414366/329c9cc3-9692-46e6-b81b-552f9458525f)

Pydroid app

![IMG_20230830_130155.jpg](https://github.com/termux/termux-app/assets/116414366/2a9f9740-0276-4def-9961-8f531dd91911)

Anyone please help me to slove this?

**Comment by sylirre** [comment]
Pydroid provides own library of python modules, that's why it works. But on Termux you use official PyPI and considering that Termux is "custom" platform, the module should be built from source.

In your case you don't have Rust compiler installed and module won't build native extensions without it. By `pkg install rust` you should get further.

**Reply by shaheen-coder** [reply]
Bro first of all thanks for helping me, I install rust pkg but also I can't install django-allauth pip

**Reply by sylirre** [reply, marked as answer]
Read the `pip` errors to find out what is missing. It is *possible* to install django-allauth.

Most of issues come from `cryptography` module.

![Screenshot_20230902-164227](https://github.com/termux/termux-app/assets/107305601/7bcbb30f-cda6-4126-bcc6-6b59b5a20044)

**Reply by shaheen-coder** [reply]
still it doesnt work i done all of the command above you mentioned

=============================DEBUG ASSISTANCE=============================

      error: can't find Rust compiler

      If you are using an outdated pip version, it is possible a prebuilt wheel is available for this package but pip is not able to install from it. Installing from the wheel would avoid the need for a Rust compiler.

      To update pip, run:

          pip install --upgrade pip

      and then retry package installation.

      If you did intend to build this package from source, try installing a Rust compiler from your system package manager and ensure it is on the PATH during installation. Alternatively, rustup (available at https://rustup.rs) is the recommended way to download and update the Rust compiler toolchain.

      This package requires Rust >=1.56.0.
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for cryptography
Failed to build cryptography
ERROR: Could not build wheels for cryptography, which is required to install pyproject.toml-based p

**Reply by sylirre** [reply]
You didn't executed all necessary commands. See screenshot. You need the `rust` package and few others.

Or try `pkg install python-cryptography` if for some reason you try to avoid installing rust.

**Reply by shaheen-coder** [reply]
Bro, really thank you so so much it now works good

**Reply by edilanejaderlan-cyber** [reply]
When does not work streamlit on pydroid
